marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -610,22 +730,44 @@ loadGist gist = do
   toJavascriptEditor $ JavascriptEditor.editorSetValue $ fromMaybe mempty javascript
   for_ actus \xml -> query _actusBlocklySlot unit (ActusBlockly.LoadWorkspace xml unit)
 
+------------------------------------------------------------
+preventAccidentalNavigation ::
+  forall m.
+  MonadAff m =>
+  SPSettings_ SPParams_ ->
+  Action ->
+  HalogenM State Action ChildSlots Void m Unit ->
+  HalogenM State Action ChildSlots Void m Unit
+preventAccidentalNavigation settings intendedAction guardedEffect = do
+  state <- H.get
+  let
+    hasUnsavedChanges =
+      maybe false identity do
+        lens <- currentLang state <#> langHasUnsavedChanges
+        pure $ state ^. lens
```

Comment:
I don't think this logic is correct. For example, if we are in the Haskell workflow, we want to check that neither the haskell editor nor the marlowe editor has unsaved changes. At this stage I think it's easiest if we check all the editors for unsaved changes, every time. Maybe in the future it will be worth adding some more logic here but for now I think that's easiest.